scripts/cnv_wdl/cnv_common_tasks.wdl

```diff
@@ -605,3 +616,122 @@ task CollectModelQualityMetrics {
+        model_files_tar_list=~{write_lines(model_tars)}
+
+        calling_configs_list=~{write_lines(calling_configs)}
+        denoising_configs_list=~{write_lines(denoising_configs)}
+        gcnvkernel_version_list=~{write_lines(gcnvkernel_version)}
+        sharded_interval_lists_list=~{write_lines(sharded_interval_lists)}
+
+        cat $calls_files_tar_list | sort -V > calls_files_tar_list.sorted
+        cat $model_files_tar_list | sort -V > model_files_tar_list.sorted
+
+        cat $calling_configs_list | sort -V > calling_configs_list.sorted
+        cat $denoising_configs_list | sort -V > denoising_configs_list.sorted
+        cat $gcnvkernel_version_list | sort -V > gcnvkernel_version_list.sorted
+        cat $sharded_interval_lists_list | sort -V > sharded_interval_lists_list.sorted
+
+        paste calls_files_tar_list.sorted model_files_tar_list.sorted calling_configs_list.sorted denoising_configs_list.sorted gcnvkernel_version_list.sorted sharded_interval_lists_list.sorted |\
+              awk '{print (NR-1)"\t"$0}' > file_sets.sorted
+        OIFS=$IFS
+        IFS=$'\t'
+        while read index calls_tar model_tar call_config denoise version intervals; do
+            mkdir -p out/CALLS_$index
+            mkdir -p out/MODEL_$index
+            tar xzf $calls_tar -C out/CALLS_$index
+            tar xzf $model_tar -C out/MODEL_$index
+            cp $call_config out/CALLS_$index
+            cp $denoise out/CALLS_$index
+            cp $version out/CALLS_$index
+            cp $intervals out/CALLS_$index
+            rm $calls_tar $model_tar $call_config $denoise $version $intervals
+
+        done < file_sets.sorted
+        IFS=$OIFS
+
+        tar c -C out . | gzip -1 > case-gcnv-postprocessing-invariants.tar.gz
+        rm -Rf out
+    >>>
+
+    runtime {
+        docker: docker
+        memory: select_first([mem_gb, 2]) + " GiB"
+        disks: "local-disk " + select_first([disk_space_gb, 150]) + if use_ssd then " SSD" else " HDD"
+        cpu: select_first([cpu, 1])
+        preemptible: select_first([preemptible_attempts, 5])
+    }
+
+    output {
+        File bundle_tar = "case-gcnv-postprocessing-invariants.tar.gz"
+    }
+}
+
+task ScatterPloidyCallsBySample {
+    input {
+      File contig_ploidy_calls_tar
+      Array[String] samples
+
+      # Runtime parameters
+      String docker
+      Int? mem_gb
+      Int? disk_space_gb
+      Boolean use_ssd = false
+      Int? cpu
+      Int? preemptible_attempts
+    }
+
+    Int num_samples = length(samples)
+    String out_dir = "calls_renamed"
+
+    command <<<
+      set -eu
+
+      # Extract ploidy calls
+      mkdir calls
+      tar xzf ~{contig_ploidy_calls_tar} -C calls/
+
+      # Archive call files by sample, renaming so they will be glob'd in order
+      sample_ids=(~{sep=" " samples})
+      for (( i=0; i<~{num_samples}; i++ ))
+      do
+        sample_id=${sample_ids[$i]}
+        sample_no=`printf %04d $i`
```

Comment:
I think the way this is originally done in the gCNV tasks is more robust to the maximum number of samples.

Reply:
Also, why is the renaming added to the ploidy calls here, but removed from the gCNV calls? Are those guaranteed to line up?

Reply:
Added dynamic scaling of the sample index padding. I don't think we need sample ids in the call tars since these are only consumed internally. The sample order will be the same between the input bam array and `SAMPLE_` subdirectories that gCNV creates, so they will line up.